src/Sarif/ExtensionMethods.cs

```diff
@@ -15,6 +15,37 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class ExtensionMethods
     {
+        /// <summary>
+        /// Attempt to reconstruct a URI, if appropriate, using Run instance 
+        /// originalUriBaseId and uriBaseId properties. If this method cannot
+        /// successfully reconstitute an absolute URI, it will return false
+        /// and populate 'resolvedUri' with the input value.
+        /// </summary>
+        /// <param name="run"></param>
+        /// <param name="uri"></param>
+        /// <param name="resolvedUri"></param>
+        /// <returns></returns>
+        public static bool TryResolveUri(this Run run, FileLocation fileLocation, out Uri resolvedUri)
+        {
+            resolvedUri = fileLocation.Uri;
+
+            // We can't restore any absolute URIs unless someone has
+            // deconstructed them using uriBaseId + originalUriBaseIds
+            if (run.OriginalUriBaseIds == null) { return false; }
+
+            if (!string.IsNullOrEmpty(fileLocation.UriBaseId) &&
+                !fileLocation.Uri.IsAbsoluteUri)
+            {
+                if (run.OriginalUriBaseIds.TryGetValue(fileLocation.UriBaseId, out Uri originalUriBaseId))
```

Comment:
>Uri [](start = 83, length = 3)

I didn't know you could declare an `out` parameter inline like this. Nice. Is it new in C# 7?